sys/net/application_layer/ng_zep/ng_zep.c

```diff
@@ -0,0 +1,1072 @@
+                               sizeof(ng_pktsnip_t *));
+                ack.type = NG_NETDEV_MSG_TYPE_EVENT;
+                ack.content.value = _EVENT_RX_STARTED;
+                msg_send_int(&ack, dev->mac_pid);
+                break;
+
+            case NG_NETAPI_MSG_TYPE_SND:
+                DEBUG("zep: NG_NETAPI_MSG_TYPE_SND\n");
+                _send(dev, (ng_pktsnip_t *)msg.content.ptr);
+                break;
+
+            case NG_NETAPI_MSG_TYPE_GET:
+                DEBUG("zep: NG_NETAPI_MSG_TYPE_GET\n");
+                opt = (ng_netapi_opt_t *)msg.content.ptr;
+                ack.type = NG_NETAPI_MSG_TYPE_ACK;
+                ack.content.value = _get(dev, opt->opt, opt->data, opt->data_len);
+                msg_reply(&msg, &ack);
+                break;
+
+            case NG_NETAPI_MSG_TYPE_SET:
+                DEBUG("zep: NG_NETAPI_MSG_TYPE_SET\n");
+                opt = (ng_netapi_opt_t *)msg.content.ptr;
+                ack.type = NG_NETAPI_MSG_TYPE_ACK;
+                ack.content.value = _set(dev, opt->opt, opt->data, opt->data_len);
+                msg_reply(&msg, &ack);
+                break;
+
+            default:
+                DEBUG("udp: received unidentified message 0x%04" PRIx16 "\n",
+                      msg.type);
+                break;
+        }
+    }
+
+    return NULL;
+}
+
+static ng_pktsnip_t *_zep_hdr_build(ng_zep_t *dev, size_t size, bool ack)
+{
+    ng_pktsnip_t *zep;
+
+    switch (dev->version) {
+        case 1:
+            DEBUG("zep: Build ZEPv1 data header in pktbuf\n");
+            zep = ng_pktbuf_add(NULL, NULL, sizeof(ng_zep_v1_hdr_t) + size,
+                                NG_NETTYPE_UNDEF);
+            break;
+
+        case 2:
+            if (ack) {
+                DEBUG("zep: Build ZEPv2 ACK header in pktbuf\n");
+                zep = ng_pktbuf_add(NULL, NULL, sizeof(ng_zep_v2_ack_hdr_t) + size,
+                                    NG_NETTYPE_UNDEF);
+            }
+            else {
+                DEBUG("zep: Build ZEPv2 data header in pktbuf\n");
+                zep = ng_pktbuf_add(NULL, NULL, sizeof(ng_zep_v2_data_hdr_t) + size,
+                                    NG_NETTYPE_UNDEF);
+            }
+
+            break;
+
+        default:
+            DEBUG("zep: malconfigured version: %" PRIu8 "\n", dev->version);
+            return NULL;
+    }
+
+    return zep;
+}
+
+static inline size_t _zep_hdr_fill_v1(ng_zep_t *dev, ng_zep_v1_hdr_t *hdr,
+                                      size_t payload_len)
+{
+    hdr->version = 1;
+    hdr->chan = dev->chan - NG_ZEP_MIN_CHANNEL;
+    hdr->dev = byteorder_htons(1);
+    hdr->lqi_mode = dev->lqi_mode;
+    hdr->lqi_val = 0xff;                /* TODO: set */
+    memset(hdr->resv, 0, sizeof(hdr->resv));
+    hdr->length = payload_len;
+
+    return sizeof(ng_zep_v1_hdr_t);
+}
+
+static size_t _zep_hdr_fill_v2_data(ng_zep_t *dev, ng_zep_v2_data_hdr_t *hdr,
```

Comment:
nitpicky: why does this function have a different naming scheme in comparison with the `_v1` one?